charms/istio-pilot/lib/charms/istio_pilot/v0/istio_gateway_info.py

```diff
@@ -50,15 +88,41 @@ def some_event_function():
 
 # Increment this PATCH version before using `charmcraft publish-lib` or reset
 # to 0 if you are raising the major API version
-LIBPATCH = 1
+LIBPATCH = 2
 
-
-RELATION_NAME = "gateway-info"
-INTERFACE_NAME = "istio-gateway-info"
+# Default relation and interface names. If changed, consistency must be kept
+# across the provider and requirer.
+DEFAULT_RELATION_NAME = "gateway-info"
+DEFAULT_INTERFACE_NAME = "istio-gateway-info"
 
 logger = logging.getLogger(__name__)
 
 
+def relation_preflight_checks(
```

Comment:
to the point of Andrew https://github.com/canonical/istio-operators/pull/237/files#r1170064533

maybe make this method private ?

Reply:
+1 on private method, not sure what you mean about the first phrase.